girder/utility/acl_mixin.py

```diff
@@ -52,16 +52,15 @@ def load(self, id, level=AccessType.ADMIN, user=None, objectId=True,
         """
         doc = Model.load(self, id=id, objectId=objectId, fields=fields, exc=exc)
 
-        if doc is not None:
+        if not force and doc is not None:
             if doc.get(self.resourceParent):
                 loadType = self.resourceColl
                 loadId = doc[self.resourceParent]
             else:
-                loadType = doc['attachedToType']
-                loadId = doc['attachedToId']
-
-        if not force and doc is not None:
-            self.model(loadType).load(loadId, level=level, user=user, exc=exc)
+                loadType = doc.get('attachedToType')
+                loadId = doc.get('attachedToId')
+            if loadType is not None and loadId is not None:
```

Comment:
I don't think the failure of this condition should silently pass. `Model.load` is a security-critical method. If something inheriting from `AccessControlMixin` fails to either specify `self.resourceColl` / `self.resourceParent` or have an `'attachedToType'` `'attachedToId'`, I think it should be an error, and the model should fail to load.